worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/BukkitPlayerBlockBag.java

```diff
@@ -129,8 +132,7 @@ public void storeBlock(BlockState blockState, int amount) throws BlockBagExcepti
                 continue;
             }
 
-            if (bukkitItem.getTypeId() != blockState.getBlockType().getLegacyId()) {
-                // TODO Fix when bukkit gets not terrible
+            if (!bukkitItem.getType().getKey().toString().equals(blockState.getBlockType().getId())) {
```

Comment:
Utility method, maybe?

`BukkitUtil`:
```java
public static boolean equals(final BlockType type, final Material material) {
  return material.getKey().equals(type.getId());
}
```

`BukkitPlayerBlockBag` (and other places):
```java
if (!BukkitUtil.equals(blockState.getBlockType(), bukkitItem.getType())) {
```

Reply:
Yeah, I'll do that. Thanks